plugins/modules/nd_backup.py

```diff
@@ -0,0 +1,164 @@
+    type: str
+  encryption_key:
+    description:
+    - The encryption_key for a backup file.
+    type: str
+  file_location:
+    description:
+    - The download path and file name for a backup.
+    type: str
+  backup_key:
+    description:
+    - The key generated by ND during creation of a backup.
+    - This key is required when querying or deleting a backup among multiple backups that have the same name.
+    - This key can be obtained by querying the backup.
+    type: str
+  state:
+    description:
+    - Use C(backup) for creating a backup of the cluster config.
+    - Use C(query) for listing all the backed up files.
+    - Use C(absent) for deleting a backup job.
+    type: str
+    choices: [ backup, query, absent ]
+    default: backup
+extends_documentation_fragment: cisco.nd.modules
+"""
+
+EXAMPLES = r"""
+- name: Create a Backup
+  cisco.nd.nd_backup:
+    name: nexus
+    encryption_key: testtest
+    file_location: ./nexus.tgz
+    state: backup
+
+- name: Query a Backup job
+  cisco.nd.nd_backup:
+    name: nexus
+    state: query
+  register: query_result
+
+- name: Query all the backup jobs
+  cisco.nd.nd_backup:
+    state: query
+  register: query_results
+
+- name: Delete a Backup job
+  cisco.nd.nd_backup:
+    name: nexus
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+from ansible.module_utils._text import to_bytes
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.nd.plugins.module_utils.nd import NDModule, nd_argument_spec, write_file
+
+
+def main():
+    argument_spec = nd_argument_spec()
+    argument_spec.update(
+        name=dict(type="str", aliases=["backup_name"]),
+        encryption_key=dict(type="str", no_log=False),
+        file_location=dict(type="str"),
+        backup_key=dict(type="str", no_log=False),
+        state=dict(type="str", default="backup", choices=["backup", "query", "absent"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "backup", ["name", "encryption_key", "file_location"]],
+            ["state", "absent", ["name"]],
+        ],
+    )
+
+    nd = NDModule(module)
+
+    name = nd.params.get("name")
+    encryption_key = nd.params.get("encryption_key")
+    backup_key = nd.params.get("backup_key")
+    file_location = nd.params.get("file_location")
+    state = nd.params.get("state")
+
+    if encryption_key is not None and len(encryption_key) < 8:
+        nd.fail_json("Please provide a minimum of 8 characters for the encryption key.")
+
+    path = "/nexus/infra/api/platform/v1/exports"
+    # The below path for GET operation is to be replaced by an official documented API endpoint once it becomes available.
+    backup_objs = nd.query_obj("/api/config/class/exports")
+    backups_info = []
+    if name:
+        backups_info = [file_dict for file_dict in backup_objs if file_dict.get("description") == name]
```

Comment:
Since you use list comprehension for creating the backups_info the empty list assignment is redundant.

```suggestion
    if name:
        backups_info = [file_dict for file_dict in backup_objs if file_dict.get("description") == name]
```